iina/FileGroup.swift

```diff
@@ -144,8 +143,7 @@ class FileGroup {
 
   func flatten() -> [String: [FileInfo]] {
     var result: [String: [FileInfo]] = [:]
-    var search: ((FileGroup) -> Void)!
-    search = { group in
+    func search(_ group: FileGroup) {
```

Comment:
> There seems to be a bug inside Swift itself which causes memory leak when there's recursive call of closure.

Is there? Give me a few more details and I'll file it on the [Swift JIRA](http://bugs.swift.org/).